kibana-reports/server/index.ts

```diff
@@ -13,8 +13,34 @@
  * permissions and limitations under the License.
  */
 
-import { PluginInitializerContext } from '../../../src/core/server';
 import { OpendistroKibanaReportsPlugin } from './plugin';
+import { schema, TypeOf } from '@kbn/config-schema';
+import {
+  PluginInitializerContext,
+  PluginConfigDescriptor,
+  HttpServerInfo,
+} from '../../../src/core/server';
+
+export const configSchema = schema.object({
+  access: schema.object({
+    port: schema.number({ defaultValue: 5601 }),
+    basePath: schema.string({ defaultValue: '' }),
+  }),
+});
+
+export type KibanaReportsPluginConfigType = TypeOf<typeof configSchema>;
+
+export const config: PluginConfigDescriptor<KibanaReportsPluginConfigType> = {
+  exposeToBrowser: {
+    access: true,
+  },
+  schema: configSchema,
```

Comment:
what does this file do besides defining `KibanaReportsPluginConfigType`? it seems it's adding port and basepath config to the kibana.yml file? shouldn't be necessary since we are not reading from yml

Reply:
This is something I forget to remove when I tried the plugin config approach. Will remove